core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -68,68 +78,161 @@ protected int run() throws Exception {
         Jenkins jenkins = Jenkins.get();
         jenkins.checkPermission(Jenkins.ADMINISTER);
 
-        PluginManager manager = jenkins.getPluginManager();
-        int result = 0;
-        boolean somePluginDisabled = false; // to ensure we restart when it needs to
-        for (String pluginName : pluginNames) {
-            switch (disablePlugin(manager, pluginName)) {
-                case DISABLED:
-                    if(!somePluginDisabled) somePluginDisabled = true;
-                    break;
-                case NOT_DISABLED_DEPENDANTS:
-                    if(result != 16) result = 16; // Custom error code to indicate one plugin, at least, can't be disabled
+        // get the strategy as an enum
+        PluginWrapper.PluginDisableStrategy strategyToUse;
+        try {
+            strategyToUse = PluginWrapper.PluginDisableStrategy.valueOf(strategy.toUpperCase());
+        } catch (IllegalArgumentException iae) {
+            throw new IllegalArgumentException(hudson.cli.Messages.DisablePluginCommand_NoSuchStrategy(strategy, String.format("%s, %s, %s", PluginWrapper.PluginDisableStrategy.NONE, PluginWrapper.PluginDisableStrategy.MANDATORY, PluginWrapper.PluginDisableStrategy.ALL)));
+        }
+
+        // disable...
+        List<PluginWrapper.PluginDisableResult> results = jenkins.pluginManager.disablePlugins(strategyToUse, pluginNames);
+
+        // print results ...
+        printResults(results);
+
+        // restart if it was required and it's necessary (some plugin was disabled in this execution)
+        restartIfNecessary(results);
+
+        // return the appropriate error code
+        return getResultCode(results);
+    }
+
+    /**
+     * Print the result of all the process
+     * @param results the list of results for the disablement of each plugin
+     */
+    private void printResults( List<PluginWrapper.PluginDisableResult> results) {
+        for (PluginWrapper.PluginDisableResult oneResult : results) {
+            printResult(oneResult, 0);
+        }
+    }
+
+    /**
+     * Print indented the arguments with the format passed beginning with the indent passed.
+     * @param indent number of spaces at the beginning.
+     * @param format format as in {@link String#format(String, Object...)}
+     * @param arguments arguments to print as in {@link String#format(String, Object...)}
+     */
+    private void printIndented(int indent, String format, String... arguments) {
+        if (indent == 0) {
+            stdout.format(format + "%n", (Object[]) arguments);
+        } else {
+            String f = "%" + indent + "s" + format + "%n";
+            String[] newArgs = new String[arguments.length + 1];
+            newArgs[0] = " "; int i = 1;
+            for (String arg: arguments ) {
```

Comment:
nit: `for (String arg: arguments ) {` => `for (String arg: arguments) {` (removal of space)